src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/rope_fusion.cpp

```diff
@@ -746,6 +746,34 @@ ov::intel_cpu::RoPEFusionQwen::RoPEFusionQwen(int split_output_id) {
             config.slice_stop = config.slice_start + config.head_cnt * config.head_size;
         }
 
+        if (pattern_map.count(reshape_special)) {
+            // check reshape_special shape correctness
+            auto reshape_special_node = pattern_map.at(reshape_special).get_node_shared_ptr();
+            auto data_shape = reshape_special_node->get_input_partial_shape(0);
+            auto reshape_shape = pattern_map.at(const_shape);
+            auto check_const = [&data_shape](Output<Node> output) -> bool {
+                auto node = std::dynamic_pointer_cast<opset1::Constant>(output.get_node_shared_ptr());
```

Comment:
It's better to use OV RTTI for casting:
```suggestion
                auto node = ov::as_type_ptr<opset1::Constant>(output.get_node_shared_ptr());
```